src/storage/local_storage/local_table.cpp

```diff
@@ -89,142 +55,53 @@ void LocalVectorCollection::prepareAppend() {
-    const std::vector<ValueVector*>& outputVectors) {
-    auto nodeGroupIdx = initializeLocalNodeGroup(nodeIDVector);
-    nodeGroups.at(nodeGroupIdx)->scan(nodeIDVector, columnIDs, outputVectors);
-}
-
-void LocalTableData::lookup(ValueVector* nodeIDVector, const std::vector<column_id_t>& columnIDs,
-    const std::vector<ValueVector*>& outputVectors) {
-    for (auto i = 0u; i < nodeIDVector->state->selVector->selectedSize; i++) {
-        auto nodeIDPos = nodeIDVector->state->selVector->selectedPositions[i];
-        auto nodeOffset = nodeIDVector->getValue<nodeID_t>(nodeIDPos).offset;
-        auto nodeGroupIdx = initializeLocalNodeGroup(nodeOffset);
-        for (auto columnIdx = 0u; columnIdx < columnIDs.size(); columnIdx++) {
-            auto columnID = columnIDs[columnIdx];
-            auto outputVector = outputVectors[columnIdx];
-            auto outputVectorPos = outputVector->state->selVector->selectedPositions[i];
-            nodeGroups.at(nodeGroupIdx)
-                ->lookup(nodeOffset, columnID, outputVector, outputVectorPos);
         }
     }
-}
-
-void LocalTableData::insert(
-    ValueVector* nodeIDVector, const std::vector<ValueVector*>& propertyVectors) {
-    KU_ASSERT(nodeIDVector->state->selVector->selectedSize == 1);
-    auto nodeGroupIdx = initializeLocalNodeGroup(nodeIDVector);
-    nodeGroups.at(nodeGroupIdx)->insert(nodeIDVector, propertyVectors);
-}
-
-void LocalTableData::update(
-    ValueVector* nodeIDVector, column_id_t columnID, ValueVector* propertyVector) {
-    auto nodeGroupIdx = initializeLocalNodeGroup(nodeIDVector);
-    nodeGroups.at(nodeGroupIdx)->update(nodeIDVector, columnID, propertyVector);
-}
-
-void LocalTableData::delete_(ValueVector* nodeIDVector) {
-    auto nodeIDPos = nodeIDVector->state->selVector->selectedPositions[0];
-    auto nodeOffset = nodeIDVector->getValue<nodeID_t>(nodeIDPos).offset;
-    auto nodeGroupIdx = StorageUtils::getNodeGroupIdx(nodeOffset);
-    if (!nodeGroups.contains(nodeGroupIdx)) {
-        return;
-    }
-    nodeGroups.at(nodeGroupIdx)->delete_(nodeIDVector);
-}
-
-node_group_idx_t LocalTableData::initializeLocalNodeGroup(ValueVector* nodeIDVector) {
-    auto nodeIDPos = nodeIDVector->state->selVector->selectedPositions[0];
-    auto nodeOffset = nodeIDVector->getValue<nodeID_t>(nodeIDPos).offset;
-    return initializeLocalNodeGroup(nodeOffset);
-}
-
-node_group_idx_t LocalTableData::initializeLocalNodeGroup(offset_t nodeOffset) {
-    auto nodeGroupIdx = StorageUtils::getNodeGroupIdx(nodeOffset);
-    if (!nodeGroups.contains(nodeGroupIdx)) {
-        nodeGroups.emplace(nodeGroupIdx, std::make_unique<LocalNodeGroup>(dataTypes, mm));
-    }
-    return nodeGroupIdx;
-}
-
-LocalTableData* LocalTable::getOrCreateLocalTableData(
-    const std::vector<std::unique_ptr<Column>>& columns, MemoryManager* mm) {
-    if (!localTableData) {
+    auto dataIdx = tableType == TableType::NODE ? 0 : direction == RelDataDirection::FWD ? 0 : 1;
```

Comment:
I see this pattern: `RelDataDirection::FWD ? 0 : 1;` many times, can we refactor to a function?